### PR TITLE
Remove clone args

### LINKED ISSUE #7457: Cleanup args clone
We should review and cleanup cloning `args` in https://github.com/TravisEz13/PowerShell/blob/92a8b9bfcf8bcf07de873f8a8759c86f239ce9a6/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs#L153

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

Close #7457.

The cleanup is needed after #7449 and #2538.

- args is never null
- We never change the args array and exclude cloning.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by TravisEz13** [comment]
@daxian-dbw Not sure why this was there in the first place.

**Review by anmenaga — CHANGES_REQUESTED** [review]
It would be good to get a [feature] test pass on this. Thanks.

Also, @daxian-dbw any ideas regarding Travis's question? I wonder the same thing as well...

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 1]
```diff
@@ -163,16 +160,11 @@ internal sealed partial class ConsoleHost
                     hostException = e;
                 }
 
-                if (args == null)
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: We should probably leave the null check in the code.

Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: Is c# convention that args is never null?

**Review by anmenaga — APPROVED** [review]

**Pull request merged**